lib/compat/wordpress-6.3/navigation-block-preloading.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Patches resources loaded by the block editor page
+ * to include Navigation posts.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Preloads requests needed for Navigation posts
+ *
+ * @param array $preload_paths    Preload paths to be filtered.
+ * @return array
+ */
+function gutenberg_preload_navigation_posts( $preload_paths, $context ) {
+
+	// Limit to the Site Editor.
+	if ( ! empty( $context->name ) && 'core/edit-site' !== $context->name ) {
+		return $preload_paths;
+	}
+
+	$navigation_rest_route = rest_get_route_for_post_type_items(
+		'wp_navigation'
+	);
+
+	// Preload the OPTIONS request for all Navigation posts request.
+	$preload_paths[] = array( $navigation_rest_route, 'OPTIONS' );
+
+	// Preload the GET request for all 'published' or 'draft' Navigation posts.
+	$preload_paths[] = array(
+		add_query_arg(
+			array(
+				'context'   => 'edit',
+				'per_page'  => '100',
+				'_locale'   => 'user',
+				// array indices are required to avoid query being encoded and not matching in cache.
+				'status[0]' => 'publish',
+				'status[1]' => 'draft',
+			),
+			$navigation_rest_route
+		),
+		'GET',
+	);
+
+	// Preload the GET request for 'published' Navigation posts only.
+	$preload_paths[] = array(
+		add_query_arg(
+			array(
+				'context'  => 'edit',
+				'per_page' => '100',
```

Comment:
Aren't `per_page` values supposed to be integers?

Reply:
I can change that 👍 although I bet they get converted to `int`.

Reply:
✅